packages/headless/src/features/generated-answer/generated-answer-analytics-actions.ts

```diff
@@ -0,0 +1,60 @@
+import {GeneratedAnswerCitation} from '../../api/generated-answer/generated-answer-event-payload';
+import {
+  AnalyticsType,
+  CustomAction,
+  SearchAction,
+  makeAnalyticsAction,
+} from '../analytics/analytics-utils';
+
+export const logRetryGeneratedAnswer = (): SearchAction =>
+  makeAnalyticsAction(
+    'generatedAnswer/logRetryGeneratedAnswer',
+    AnalyticsType.Search,
+    (client) => client.makeRetryGeneratedAnswer()
+  );
+
+export const logOpenGeneratedAnswerSource = (
+  citation: GeneratedAnswerCitation
+): CustomAction =>
+  makeAnalyticsAction(
+    'generatedAnswer/logOpenGeneratedAnswerSource',
+    AnalyticsType.Custom,
+    (client, state) => {
+      const generativeQuestionAnsweringId =
+        state.search?.response?.extendedResults
+          ?.generativeQuestionAnsweringId ?? '';
+      return client.makeOpenGeneratedAnswerSource({
+        generativeQuestionAnsweringId,
+        permanentId: citation.permanentid,
+        id: citation.id,
+      });
+    }
+  );
+
+export const logLikeGeneratedAnswer = (): CustomAction =>
+  makeAnalyticsAction(
+    'generatedAnswer/logLikeGeneratedAnswer',
```

Comment:
Looks like all other analytics actions have this format: `'analytics/generatedAnswer/likeGeneratedAnswer'` 

Don't know if it matters a lot to be consistent or not in this

Reply:
Consistent is better